### DISCUSSION TITLE
Install Google Cloud SDK with Homebrew on Linux

### DISCUSSION THREAD

**Discussion by damlys** [Homebrew/discussions#3594] [Writing Formulae/Casks]
[Google Cloud SDK](https://cloud.google.com/sdk) can be installed with Homebrew only on macOS devices with the following command:

```
$ brew install --cask google-cloud-sdk
```

I think it would be cool to be able do the same on [Linux](https://docs.brew.sh/Homebrew-on-Linux) devices:

```
$ brew install google-cloud-sdk
```

The same way as for AWS/Azure CLIs:

```
$ brew install awscli
$ brew install azure-cli
```

Unfortunately there is only the [GCP SDK cask](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/google-cloud-sdk.rb). Someone think [GCP SDK formula](https://github.com/Homebrew/homebrew-core) would be useful too? 😄 

---

Following `Dockerfile` produces error 😢 

```
FROM homebrew/brew:latest

RUN brew install google-cloud-sdk
```

This one is not so pretty 🙈 

```
FROM homebrew/brew:latest

RUN apt-get update && apt-get install --yes --no-install-recommends \
  apt-transport-https \
  ca-certificates \
  gnupg \
&& apt-get clean && rm -rf /var/lib/apt/lists/*

RUN echo 'deb [signed-by=/usr/share/keyrings/cloud.google.gpg] http://packages.cloud.google.com/apt cloud-sdk main' | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list \
&& curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg  add - \
&& apt-get update  && apt-get install --yes --no-install-recommends \
  google-cloud-cli \
&& apt-get clean && rm -rf /var/lib/apt/lists/*
```

**Comment by Bo98** [comment]
One requirement for being in homebrew-core would be that it is open source. Every formulae we ship is built from source, even if what you end up downloading is a bottle prebuilt by our CI.

**Reply by alebcay** [reply]
Downloading the prebuilt tarball from Google's site, it contains this `LICENSE`:
```
The Google Cloud CLI and its source code are licensed under Apache
License v. 2.0 (the "License"), unless otherwise specified by an alternate
license file.

You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Note that if you use the Google Cloud CLI with any Google Cloud Platform
products, your use is additionally going to be governed by the license agreement
or terms of service, as applicable, of the underlying Google Cloud Platform
product with which you are using the Google Cloud CLI. For example, if you are
using the Google Cloud CLI with Google App Engine, your use would additionally
be governed by the Google App Engine Terms of Service.

This also means that if you were to create works that call Google APIs, you
would still need to agree to the terms of service (usually, Google's
Developer Terms of Service at https://developers.google.com/terms) for those
APIs separately, as this code does not grant you any special rights to use
the services.
```

IANAL but I don't think the latter two paragraphs here infringes on the preceding `Apache-2.0` license in any way; I think it's just clarifying that the right to use of the SDK does not grant a right to GCP services.

~~I also note the Google Cloud SDK download site has OS/architecture-specific downloads. I'm not sure what actually differs here between platforms since it's written in Python, so taking a deeper look.~~

There are some platform-specific binaries included in the package, some of which don't appear to be open-source (e.g. `anthoscli`). We could try and delete the binaries that we can't build from source, but that sounds like a tedious process. Some packagers seem to take this route though (FreeBSD ports, AUR).

I think there's still value in a formula to provide this on Linux platforms; however, unless we find a way to build some of these binaries from source, I don't see a good way to include this (with the same level of functionality as the current cask) in the core tap.
